pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/fixtures/HybridRecordReceiverFixture.scala

```diff
@@ -1,34 +1,119 @@
+import uk.ac.wellcome.storage.store.{
+  HybridIndexedStoreEntry,
+  HybridStore,
+  HybridStoreEntry
+}
+import uk.ac.wellcome.storage.fixtures.S3Fixtures.Bucket
+import uk.ac.wellcome.storage.store.memory.{
+  MemoryStore,
+  MemoryStreamStore,
+  MemoryTypedStore
+}
+import uk.ac.wellcome.models.transformable.SierraTransformable._
+import uk.ac.wellcome.storage.streaming.Codec._
+import uk.ac.wellcome.storage.streaming.InputStreamWithLengthAndMetadata
+import java.io.ByteArrayInputStream
+
+trait HybridRecordReceiverFixture extends BigMessagingFixture with SNS {
+
+  val memoryIndexedStore =
+    new MemoryStore[
+      Version[String, Int],
+      HybridIndexedStoreEntry[String, EmptyMetadata]](Map.empty)
+
+  implicit val memoryStreamStore = MemoryStreamStore[String]()
+
+  implicit val hybridStore =
+    new HybridStore[
+      Version[String, Int],
+      String,
+      SierraTransformable,
+      EmptyMetadata] {
+      override implicit val indexedStore = memoryIndexedStore
+      override implicit val typedStore
+        : MemoryTypedStore[String, SierraTransformable] =
+        new MemoryTypedStore[String, SierraTransformable](Map.empty)
+      override def createTypeStoreId(id: Version[String, Int]): String =
+        s"${id.id}/${id.version}"
+    }
 
-trait HybridRecordReceiverFixture extends Messaging with SNS {
   def withHybridRecordReceiver[R](
     topic: Topic,
     bucket: Bucket,
     snsClient: AmazonSNS = snsClient
-  )(testWith: TestWith[HybridRecordReceiver, R])(
-    implicit objectStore: ObjectStore[SierraTransformable]): R =
-    withMessageWriter[TransformedBaseWork, R](
-      bucket,
-      topic,
-      writerSnsClient = snsClient) { messageWriter =>
-      val recordReceiver = new HybridRecordReceiver(
-        messageWriter = messageWriter,
-        objectStore = objectStore
-      )
-
-      testWith(recordReceiver)
+  )(testWith: TestWith[HybridRecordReceiver[SNSConfig], R]): R =
+    withSqsBigMessageSender[TransformedBaseWork, R](bucket, topic, snsClient) {
+      msgSender =>
+        val recordReceiver = new HybridRecordReceiver(
+          msgSender = msgSender,
+          store = hybridStore
+        )
+
+        testWith(recordReceiver)
     }
+
+  def createHybridRecordNotificationWith(
+    sierraTransformable: SierraTransformable,
+    version: Int = 1): NotificationMessage = {
+
+    val hybridRecord = createHybridRecordWith(
+      sierraTransformable,
+      version = version
+    )
+    createNotificationMessageWith(
+      message = hybridRecord
+    )
+  }
+
+  def createHybridRecordWith(
+    sierraTransformable: SierraTransformable,
+    version: Int = 1,
+    id: String = Random.alphanumeric take 10 mkString): HybridRecord = {
+
+    hybridStore.put(Version(id, version))(
+      HybridStoreEntry(sierraTransformable, EmptyMetadata()))
+    HybridRecord(
+      id = id,
+      version = version,
+      location = ObjectLocation("namespace", "path")
```

Comment:
Don’t you want to use the location you get from the hybridStore.put()?

(That does return the location, right?)

Reply:
`hybridStore.put()` will return a `Version[String, Int]` (which is what can be used to access the object again from the store).

`location` is used to encode the location in S3 (as received from the adaptor), but these tests aren't using a dynamo/S3 hybrid store store (uses in memory equivilent), so it is kind of irrelevant what is in that field.

Does that make sense?

Reply:
It's a bit confusing, so will add some more clarification:

* `HybridRecord` is the message received from the adaptor (bad naming probably?)

* The  old code used the `location` field directly to access S3

* In the discussion the other day we decided we should just access the hybrid store in the standard way (using `Version[String, Int]` rather  than going directly to S3

* At some stage we can change the message put on the queue by the adaptor to just be `Version[String, Int]`, but for now we have to put something in the `location` field even though it is ignored / irrelevant